src/test/java/com/icthh/xm/ms/otp/repository/UaaRepositoryTest.java

```diff
@@ -0,0 +1,94 @@
+package com.icthh.xm.ms.otp.repository;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.icthh.xm.ms.otp.OtpApp;
+import com.icthh.xm.ms.otp.config.Constants;
+import com.icthh.xm.ms.otp.config.SecurityBeanOverrideConfiguration;
+import com.icthh.xm.ms.otp.config.tenant.WebappTenantOverrideConfiguration;
+import io.github.jhipster.config.JHipsterProperties;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestTemplate;
+import org.thymeleaf.util.StringUtils;
+
+import java.util.Map;
+import java.util.UUID;
+
+import static com.icthh.xm.ms.otp.config.Constants.ACCESS_TOKEN;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+@RunWith(SpringRunner.class)
+@WithMockUser(authorities = {"SUPER-ADMIN"})
+@SpringBootTest(classes = {
+    SecurityBeanOverrideConfiguration.class,
+    OtpApp.class,
+    WebappTenantOverrideConfiguration.class
+})
+public class UaaRepositoryTest {
+
+    @Autowired
+    UaaRepository uaaRepository;
+
+    @Qualifier("internalRestTemplate")
+    @MockBean
+    private RestTemplate restTemplate;
+
+
+    @Test
+    public void oauthTokenShouldBeRetrievedFromUaaTest() {
+
+        when(restTemplate.exchange(anyString(), eq(HttpMethod.POST), any(HttpEntity.class), any(ParameterizedTypeReference.class)))
```

Comment:
not mock restTemplate
better use 
org.springframework.test.web.client.MockRestServiceServer

Reply:
RestGatewaySupport gateway = new RestGatewaySupport();
        gateway.setRestTemplate(restTemplate);
        mockServer = MockRestServiceServer.createServer(gateway);